docs/core/testing/unit-testing-mstest-runner-runsettings.md

```diff
@@ -0,0 +1,154 @@
+
+|Node|Description|Reason / workaround |
+|-|-|-|
+|**MaxCpuCount**|This setting controls the level of parallelism on process-level. Use 0 to enable the maximum process-level parallelism.| When MSTest runner is used with MSBuild, this option is [offloaded to MSBuild]. When a single executable is run, this option has no meaning for MSTest runner.
+|**ResultsDirectory**|The directory where test results are placed. The path is relative to the directory that contains .runsettings file.| Use the commandline option `--results-directory` to determine the directory where the test results are going to be placed. If the specified directory doesn't exist, it's created. The default is TestResults in the directory that contains the test application.
+|**TargetFrameworkVersion**| This setting defines the framework version, or framework family to use to run tests.| This option is ignored. The `<TargetFramework>` or `<TargetFrameworks>` MSBuild property determines the target framework of the application. The tests are hosted in the final application.
+|**TargetPlatform**|This setting defines the architecture to use to run tests. | `<RuntimeIdentifier>` determines the architecture of the final application that hosts the tests.
+|**TreatTestAdapterErrorsAsWarnings**|Suppresses test adapter errors to become warnings. | MSTest runner allows only one type of tests to be run from a single assembly, and failure to load the test framework or other parts of infrastructure will become an unskippable error, because it signifies that some tests could not be discovered or run.
+|**TestAdaptersPaths**| One or more paths to the directory where the TestAdapters are located| MSTest runner does not use the concept of test adapters and does not allow dynamic loading of extensions unless they are part of the build, and are registered in Program.cs, either automatically via build targets or manually. |
+|**TestCaseFilter**| A filter to limit tests which will run. | To filter tests use `--vstest-filter` commandline option.
+|**TestSessionTimeout**|Allows users to terminate a test session when it exceeds a given timeout.| There is no alternative option. |
+|**DotnetHostPath**|Specify a custom path to dotnet host that is used to run the testhost. | MSTest runner is not doing any additional resolving of dotnet. It depends fully on how dotnet resolves itself, which can be controlled by environment variables such as [`DOTNET_HOST_PATH`](https://learn.microsoft.com/dotnet/core/tools/dotnet-environment-variables#dotnet_host_path)
+|**TreatNoTestsAsError**| Exit with non-zero exit code when no tests are discovered. | MSTest runner will error by default when no tests are discovered or run in a test application. You can set how many tests you expect to find in the assembly by using `--minimum-expected-tests` command line parameter, which defaults to 1.
+
+DataCollectionRunSettings?
+
+LoggerRunSettings?
+
+MSTest?
```

Comment:
MSTest settings are handled by MSTest and so they are all supported. You can link https://learn.microsoft.com/visualstudio/test/configure-unit-tests-by-using-a-dot-runsettings-file?view=vs-2022#mstest-element

Reply:
does this include AssemblyResolution ? Should that be supported, maybe we want to deprecate it in MSTest runner in favor of deprecating the custom assembly resolver?

Reply:
This is indeed also supported (even in runner mode). The deprecation is a good suggestion but this might be something we add later on. There is nothing with the runner that's preventing this (same as app domains etc...).

WDYT @MarcoRossignoli

Reply:
okay, keeping as is.